be/src/storage/lake/delta_writer.cpp

```diff
@@ -271,7 +270,6 @@ inline Status DeltaWriterImpl::flush_async() {
                 }
                 if (_tablet_manager->in_writing_data_size(_tablet_id) > _immutable_tablet_size) {
                     _is_immutable.store(true, std::memory_order_relaxed);
-                    _tablet_manager->remove_in_writing_data_size(_tablet_id);
```

Comment:
when will the _tablet_id removed from the `_tablet_in_writing_size` in TabletManager?

Reply:
when shard move to other be

Reply:
what if the shard is never migrated?

Reply:
It will be cached until the node is restarted, but the memory occupied by this data structure is limited, because the automatically generated bucket needs to be large enough (such as 1G) to generate a new one, so there will not be many

Reply:
may not necessary agree with this. be prepared that be/cn is expected to be long running for months or year without restart even it may not be possible for now.

Reply:
this is std::unordered_map<int64_t, int64_t>, we assume each record cost 32bytes memory, 100k tablet in this BE use 3.2MB memory of this be. 100k tablet * 1G = 100TB. 
This means that this BE can manage 100TB of data and only consumes 3.2MB of memory. Is this a big problem?

Reply:
ok. I would say 100k is relatively small. But I will not insist here. Just put some TODO or FIXME comments about this unordered_map, the element in it is not cleaned at all. When something crazy happens, it may reminder something when looking into the issues.

We are experience too many crazy things when there "should not be a problem" but actually take down the system and we now add boundaries and limitations. Lesson should be learned as always.

Reply:
This map will only store tablets managed on this be, so there is not much difference in other tablet meta in memory. If the tablets managed by this node are really large, I don't think this will become a bottleneck in memory consumption.

Reply:
It doesn't make the issue itself gone away by comparing it to a more serious issue.